log4j-core/src/main/java/org/apache/logging/log4j/core/config/plugins/util/ResolverUtil.java

```diff
@@ -473,8 +469,10 @@ protected void addIfMatching(final Test test, final String fqn) {
                     resourceMatches.add(url.toURI());
                 }
             }
-        } catch (final Throwable t) {
-            LOGGER.warn("Could not examine class {}", fqn, t);
+        } catch (final LinkageError e) {
+            LOGGER.debug("Could not scan class {}, since an optional dependency is missing.", fqn, e);
+        } catch (final Exception e) {
+            LOGGER.warn("Could not scan class {}.", fqn, e);
```

Comment:
```suggestion
        } catch (final LinkageError error) {
            LOGGER.debug("Could not scan class `{}`, since an optional dependency is missing.", fqn, error);
        } catch (final Exception error) {
            LOGGER.warn("Could not scan class `{}`.", fqn, error);
```

This will still log `ClassNotFoundException` at `WARN`, won't it? Shall we instead have a single `catch` block and check for the existence of `ClassNotFoundException` in the causal chain?

I am actually in favor of **not** creating a whitelist here and instead setting the logging level to `DEBUG` for all `Exception`s.

Reply:
Fixed in https://github.com/apache/logging-log4j2/pull/2836/commits/939de777cd047753b393365506cbe54e34a7af9b